scripts/sct_utils.py

```diff
@@ -22,13 +22,104 @@
 import subprocess
 import re
 from sys import stdout
+import logging
 
 import glob
 import shutil
 
+
 # TODO: under run(): add a flag "ignore error" for isct_ComposeMultiTransform
 # TODO: check if user has bash or t-schell for fsloutput definition
 
+"""
+Basic logging setup for the sct
+set SCT_LOG_LEVEL and SCT_LOG_FORMAT in ~/.sctrc to change the sct log
+format and level
+"""
+
+
+log = logging.getLogger('sct')
```

Comment:
I'm not sure about this. You have duplicated the contents of the function `start_stream_logger`.
I will try to refactor the initiation of the logger to make it more customizable.

Reply:
hi, are there any updates about this issue?

Reply:
I will submit a patch tonight. But I have not finished reviewing the PR.

Reply:
The `start_stream_logger` just does that, it start the logger, I get the logger at the root of `sct_utils` so we can use the `sct_utils.log.info()` that same way in all modules. 

Then, the `stream_handler` in initialise at the root  so its easier to pause the logging with  `pause_stream_logger`

I think is a good trade off between code usability and code "cleaniness".

Reply:
any update? has the issue been fixed? can we test?

Reply:
@jcohenadad So far, the code looks sound and for the first iteration, it works and can be tested.

Reply:
the issue I reported 12 days ago is still there. 

the following was tested on commit 1fdd46aea1831efe3e349944c0d311b33850a71b:
~~~
sct_propseg -i t2.nii.gz -c t2

--
Spinal Cord Toolbox (poq_issue_1409_log/1fdd46aea1831efe3e349944c0d311b33850a71b)
Running /Users/julien/code/sct/scripts/sct_propseg.py -i t2.nii.gz -c t2
Detecting the spinal cord using OptiC
Generate output files...
Finished. To view results, type:
fslview t2_int16.nii.gz &
isct_propseg -i "/Users/julien/sct_example_data/t2/t2.nii.gz" -t t2 -o "./" -verbose -init-centerline ./t2_centerline_optic.nii.gz -centerline-binary
Initialization - using given centerline
Total propagation length = 236.212 mm

Segmentation finished. To view results, type:
fslview /Users/julien/sct_example_data/t2/t2.nii.gz ./t2_seg.nii.gz &
Check consistency of segmentation...

Create temporary folder...
tmp.segmentation.nii.gz
Get dimensions of data...
64 x 320 x 320 x 1
Change orientation...
Generate output files...
WARNING: File tmp.segmentation_RPI.nii.gz already exists. Deleting it.
Finished. To view results, type:
fslview tmp.segmentation_RPI.nii.gz &
tmp.centerline.nii.gz
Get dimensions of data...
64 x 320 x 320 x 1
Change orientation...
Generate output files...
WARNING: File tmp.centerline_RPI.nii.gz already exists. Deleting it.
Finished. To view results, type:
fslview tmp.centerline_RPI.nii.gz &
Get data dimensions...
tmp.segmentation_RPI_c.nii.gz
Get dimensions of data...
64 x 320 x 320 x 1
Change orientation...
rm -f tmp.segmentation_RPI_c_LPI.nii.gz
Generate output files...
WARNING: File /Users/julien/sct_example_data/t2/t2_seg.nii.gz already exists. Deleting it.
Finished. To view results, type:
fslview /Users/julien/sct_example_data/t2/t2_seg.nii.gz &
Remove temporary files...
WARNING: File t2_seg.nii.gz already exists. Deleting it.
fatal: Not a git repository (or any of the parent directories): .git
Sucessfully generated the QC results in /Users/julien/qc_data/qc_results.json
INFO:sct:Sucessfully generated the QC results in /Users/julien/qc_data/qc_results.json
Use the following command to see the results in a browser:
INFO:sct:Use the following command to see the results in a browser:
sct_qc -folder /Users/julien/qc_data
INFO:sct:sct_qc -folder /Users/julien/qc_data
Done! To view results, type:
INFO:sct:Done! To view results, type:
fslview t2.nii.gz t2_seg.nii.gz -l Red -b 0,1 -t 0.7 &
INFO:sct:fslview t2.nii.gz t2_seg.nii.gz -l Red -b 0,1 -t 0.7 &
~~~

Reply:
Have you done a full install? It seems that I cannot reproduce the error here.

Reply:
i did not do a full install, as this was not mentioned at the beginning of the PR. In the future, please mention if full install is required (or partial, -m -b or -d depending on the case). Thanks. Testing it now

Reply:
ok, tested and I don't see the duplicate anymore. However, the text on the last line (see below) should appear in green as was the case before:
~~~
fslview t2.nii.gz t2_seg.nii.gz -l Red -b 0,1 -t 0.7 &
~~~

Reply:
I overefractor `sct_propseg` a bit here and transformed printv with color into a full sct.log.info(). I was desperate to fix the bug, that when I realized that I needed to do a `install_sct`...
I can revert to color.

Reply:
Reverted to color in latest commit 2403aff